blazar/db/sqlalchemy/api.py

```diff
@@ -2073,12 +2074,15 @@ def _resource_property_get_or_create(session, resource_type, capability_name):
 
     if resource_property:
         return resource_property
-    else:
+    elif capability_name not in FORBIDDEN_EXTRA_CAPABILITY_NAMES:
```

Comment:
I think this logic may be more parseable if it uses the early exit pattern. We know enough information before the resource properties are even fetched to make a decision:

```python
if capability_name in FORBIDDEN_EXTRA_CAPABILITY_NAMES:
    raise db_exc.BlazarDBForbiddenExtraCapability(
            property_name=capability_name)
```